### PR TITLE
feat: Export test helpers for Select, MultiSelect and TreeSelectFields

### PR DESCRIPTION
**Pull request by bdow**
Removes the existing SelectField and MultiSelectField mocks that were extra overhead when adding new features to the actual components.

### CODE REVIEW

File: src/utils/rtl.tsx
Comment (stephenh) [inline src/utils/rtl.tsx:206]: Question: Can we scope this to nodes within the `listbox`?

File: src/utils/rtl.tsx
Comment (stephenh) [inline src/utils/rtl.tsx:244]: Question: If we have two `SelectField`s on the page, is this going to bring back selections from both select fields?

File: src/utils/rtl.tsx
Comment (stephenh) [inline src/utils/rtl.tsx:280]: Fwiw once the React 18 bump re-lands (I'm waiting on zgavin to land a backend PDF PR) I'm planning on de-lazy-ing all of our `r.foo`s to do immediate lookups, so we won't need the `r.foo()` anymore. Looking forward to it!

File: src/utils/rtl.test.tsx
Comment (stephenh) [inline src/utils/rtl.test.tsx:25]: Question: Fwiw what do you think about "Two" instead of "2"? I think that's how our current DSL works, for native selects, so will be less code to change/update to use the id instead? Or am I wrong about that...

File: src/utils/rtl.test.tsx
Comment (stephenh) [inline src/utils/rtl.test.tsx:84]: Praise: neat test!

File: src/utils/rtl.test.tsx
Comment (bdow) [reply src/utils/rtl.test.tsx:25]: You can do both! If you want to select by the label or value of the element you can. 
But you're right, selecting by the label is how we currently do it. But I kind of liked giving the user the option of choosing how they'd want to do it.

File: src/utils/rtl.tsx
Comment (bdow) [reply src/utils/rtl.tsx:244]: It should only return this selectfield's options, as the other select fields shouldn't be opened... but might be safer to query from the listbox rather than the body here. I'll update that.

File: src/utils/rtl.tsx
Comment (bdow) [reply src/utils/rtl.tsx:206]: Yes, we can. Makes sense to do. I will update.

File: src/utils/rtl.test.tsx
Comment (stephenh) [reply src/utils/rtl.test.tsx:25]: Ah nice! Agreed supporting both sgtm.

File: src/utils/rtl.tsx
Comment (stephenh) [reply src/utils/rtl.tsx]: Oh I see, sure only one will be open at once, but yep thanks for updating!

**Review by stephenh — APPROVED** [review]

**Pull request merged**
